subprojects/testing-jvm/src/main/java/org/gradle/api/tasks/testing/TestReport.java

```diff
@@ -47,9 +48,7 @@
  * Generates an HTML test report from the results of one or more {@link Test} tasks.
  */
 @DisableCachingByDefault(because = "Not made cacheable, yet")
-public class TestReport extends DefaultTask {
-    private File destinationDir;
-    private ConfigurableFileCollection resultDirs = getObjectFactory().fileCollection();
+public abstract class TestReport extends DefaultTask {
```

Comment:
Although it would be nice to do this, we can't break compatibility, so this cannot be made abstract.